probcomp/metaprob
# Move tutorial to project root

## What does this do?

* Fixes #64. Implements "Option A", moving the tutorial out of `src/`. This prepares us to move it to another repository entirely per conversations in Slack.
* Adds an explicit dependency on `lein-jupyter` to `project.clj`.
* Loads `.js` files as resources. Typically source paths contain only Clojure source files. Other files that the program will be loading at runtime are included as resources. For more information about resources see [Oracle's documentation](https://docs.oracle.com/javase/8/docs/technotes/guides/lang/resources.html).

## What are the pros and cons of doing this?

### Pros

* Fixes #64.
* Explicit dependency on `lein-jupyter`  makes it possible to work on `metaprob.tutorial.jupyter` outside of Jupyter (e.g. from a REPL started with `lein repl`).

### Cons

* Makes project directory tree more complicated.

## How do I test this?

1. Verify that all the Clojure files in `src/` can be loaded:
```
clj -Srepro \
    -Sdeps "{:deps {org.clojure/tools.namespace {:mvn/version \"0.2.11\"}}}" \
    -e "(require '[clojure.tools.namespace.repl :refer [refresh]])
        (refresh)"
```
2. Verify that the tutorial still works. Launch it by following the instructions in `tutorial/README.md` then evaluate the first few cells.
